frontend/app/src/features/room/components/code/Collaborator.tsx

```diff
@@ -0,0 +1,103 @@
+import { Box, Text, Button, HStack, VStack, Spinner } from "@chakra-ui/react";
+import { Dropdown } from "@/components/Dropdown";
+import { useEffect, useState } from "react";
+import io, { Socket } from "socket.io-client";
+import { CodeEditor, QuestionDetails } from "..";
+import { LANGUAGES } from "../../constants/languages";
+import { SingleValue } from "chakra-react-select";
+import { LanguageSupport } from "node_modules/@codemirror/language/dist";
+
+interface CollaboratorProps {
+  roomId: string;
+}
+
+export const Collaborator = ({ roomId }: CollaboratorProps) => {
+  const [renderQuestion, setRenderQuestion] = useState(true);
+  const [socket, setSocket] = useState<Socket | null>(null);
+  const [currentLanguage, setCurrentLanguage] = useState(LANGUAGES.Python)
+
+  useEffect(() => {
+    const connectSocket = io("http://localhost:8005", {
+      query: {
+        roomId: roomId
+      }
+    });
+    setSocket(connectSocket);
+    return () => {
+      if (connectSocket) {
+        connectSocket.disconnect();
+        setSocket(null);
+      }
+    };
+  }, [roomId]);
+
+  const handleLanguageChange = (e: SingleValue<{
+    label: string;
+    value: LanguageSupport;
+  }>) => {
+    setCurrentLanguage(e?.value ?? LANGUAGES.Python)
```

Comment:
use the abstracted default lang constant here

Reply:
Resolved [dee539f](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g03/pull/18/commits/dee539f42bf5a8c51844ffde801e94933b1995dc)